File: sdk/dart/lib/src/email_mfa_v2.dart
import 'http_client.dart';

/// EmailMfaV2: email-based MFA management
class EmailMfaV2Resource {
  final HttpClient httpClient;
  EmailMfaV2Resource(this.httpClient);

  /// GET /api/email_mfa_v2/status/{user_type}/{user_id} - Get MFA status
  Future<Map<String, dynamic>> getMfaStatus([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/email_mfa_v2/status/{user_type}/{user_id}', params);
  }

  /// POST /api/email_mfa_v2/send - Send one-time code
  Future<Map<String, dynamic>> sendCode([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/email_mfa_v2/send', params);
  }

  /// POST /api/email_mfa_v2/verify - Verify one-time code
  Future<Map<String, dynamic>> verifyCode([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/email_mfa_v2/verify', params);
  }

  /// GET /api/email_mfa_v2/list/{user_type}/{user_id} - List active codes
  Future<Map<String, dynamic>> listActiveCodes([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/email_mfa_v2/list/{user_type}/{user_id}', params);
  }

  /// POST /api/email_mfa_v2/revokeall - Revoke all active codes
  Future<Map<String, dynamic>> revokeAllCodes([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/email_mfa_v2/revokeall', params);
  }

  /// POST /api/email_mfa_v2/disable - Disable MFA entirely
  Future<Map<String, dynamic>> disableMfa([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/email_mfa_v2/disable', params);
  }

}

File: sdk/rust/src/email_mfa_v2.rs
//! EmailMfaV2: email-based MFA management

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct EmailMfaV2Resource {
    http: Arc<HttpClient>,
}

impl EmailMfaV2Resource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/email_mfa_v2/status/{user_type}/{user_id} - Get MFA status
    pub async fn get_mfa_status(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/email_mfa_v2/status/{user_type}/{user_id}", None).await
    }

    /// POST /api/email_mfa_v2/send - Send one-time code
    pub async fn send_code(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/email_mfa_v2/send", None).await
    }

    /// POST /api/email_mfa_v2/verify - Verify one-time code
    pub async fn verify_code(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/email_mfa_v2/verify", None).await
    }

    /// GET /api/email_mfa_v2/list/{user_type}/{user_id} - List active codes
    pub async fn list_active_codes(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/email_mfa_v2/list/{user_type}/{user_id}", None).await
    }

    /// POST /api/email_mfa_v2/revokeall - Revoke all active codes
    pub async fn revoke_all_codes(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/email_mfa_v2/revokeall", None).await
    }

    /// POST /api/email_mfa_v2/disable - Disable MFA entirely
    pub async fn disable_mfa(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.post("/api/email_mfa_v2/disable", None).await
    }

}
